erpnext/accounts/report/gross_profit/gross_profit.py

```diff
@@ -106,11 +106,14 @@ def process(self):
 		self.grouped = {}
 		self.grouped_data = []
 
+		currency_precision = cint(frappe.db.get_default("currency_precision")) or 3
```

Comment:
Maybe set these as global constants in the file? You won't need to pass it as an argument.